raft-tech/TANF-app
# Infinite FETCH_STTS

## Summary of Changes
- Updated render/request cycle to manage number of requests sent state to avoid infinitely sending requests.
- Added new error modal to inform user there are issues and that they should log out/log in to see if the issue persists.
Pull request closes #2701 

## How to Test
- Remove line 99 from `docker-compose.yml` in the backend folder.
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up
```

1. Log in and get to the `Request Access` form. 
2. You should immediately see the modal displayed in the image below.
3. Sign out and make sure you can sign out
4. Sign back in and verify that you see the error modal again since the stt list still hasnt been populated in the db.
5. exec into the backend container and run `python manage.py populate_stts`
6. Go back to the `Request Access` form and verify that you don't get the error modal.

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] FETCH_STTS should be executed a finite amount of times no matter the response.
- [x] If FETCH_STTS continuously has no response, an error indicating to the user the issue should be thrown.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
